### PR TITLE
01-ast-detector by Denis Lochmelis

### LINKED ISSUE #11: External Receiver is Used
The detector should highlight all the external receivers defined within the project (see: `AstReceiver` node and its `selector` field).

## Context

Contracts in TON communicate with each other using messages. This model resembles the [actor model](https://en.wikipedia.org/wiki/Actor_model). The contract can `send` messages to any other contract and receive using the defined receiver functions.

Tact supports two kinds of message receiver functions: internal and external. External messages [are different](https://docs.tact-lang.org/book/external#how-external-messages-are-different) from internal messages: the contract has to pay gas for receiving messages, and there are some limits in gas processing.

Thus, the logic implemented in external messages may require extra attention from an auditor.

## Examples
```
external("Check Timeout") { // Highlighted by the detector
  acceptMessage();
}
```

### PR DESCRIPTION
**Pull request by DLochmelis33**
Closes #11

### CODE REVIEW

**Review by jubnzv — CHANGES_REQUESTED** [review]
Please restore the example detector.

File: assignments/1-ast-detectors/test/contract.tact [hunk 1]
```diff
@@ -0,0 +1,5 @@
+contract Test {
+    // i only know how to write this single one based on https://docs.tact-lang.org/book/external#external-receivers
```
Comment (jubnzv) [inline assignments/1-ast-detectors/test/contract.tact:2]: A few more possible examples: https://github.com/tact-lang/tact/blob/b38457330dd34190367a036983ca7ef225128ae3/src/test/contracts/case-receive.tact

Comment (jubnzv) [reply assignments/1-ast-detectors/test/contract.tact:2]: A would also include a negative tests: just introduce an internal receiver to ensure the detector doesn't highlight it.

Comment (DLochmelis33) [reply assignments/1-ast-detectors/test/contract.tact]: :ok_hand:

**Comment by DLochmelis33** [comment]
Btw, the CI did not run for some reason

**Comment by jubnzv** [comment]
> Btw, the CI did not run for some reason

It should not for the first-time contributors.

**Review by jubnzv — APPROVED** [review]
LGTM

**Pull request merged**